### ISSUE TITLE
Upgrade request-promise dependency to 2.x

### ISSUE THREAD

**Issue by blimmer** [chiefy/vaulted#79]
1.x outputs this error on `npm install`:

```
└─┬ vaulted@3.0.0
  ├─┬ confab-addons@0.1.0
  │ ├─┬ confab@0.1.0
  │ │ └── object-assign@4.0.1
  │ ├── js-yaml@3.5.3
  │ └── lodash@3.10.1
  ├── js-yaml@3.5.3
  ├── lodash@3.10.1
  ├─┬ request-debug@0.2.0
  │ └── stringify-clone@1.1.1
  └─┬ request-promise@1.0.2
    ├── bluebird@2.10.2
    ├─┬ cls-bluebird@1.0.1
    │ └── shimmer@1.1.0
    ├── UNMET PEER DEPENDENCY continuation-local-storage@~3
    └── lodash@3.10.1
```

```
UNMET PEER DEPENDENCY continuation-local-storage@~3
```

The issue is reported [here](https://github.com/request/request-promise/issues/70) and supposedly fixed in 2.x.

**Comment by kenjones-cisco** [comment]
Thanks! We have no dependency on continuation-local-storage so the change should be pretty straight forward. Thanks! :+1:

**Comment by blimmer** [comment]
Yep - opened #80 to run tests - having some problems getting docker up and running to test locally.

**Comment by kenjones-cisco** [comment]
Resolved via #80
